### ISSUE TITLE
_GMFILE_ and _GMFUNCTION_ macros turn weird when passed to string(...)

### ISSUE THREAD

**Issue by Alphish** [YoYoGames/GameMaker-Bugs#6709]
### Description
When `_GMFILE_` and `_GMFUNCTION_` macros are printed on their own, they correctly show the file and the function the macro occurrence is located.

However, when they are passed to the `string` function, they turn into `@@FILENAME@@` and `@@FUNC@@` strings. It might be related to some compile-time evaluation of the `string` function that was introduced recently.

### Steps To Reproduce
Create an empty project and a script with the following content:
```gml
function panic()
{
    show_debug_message("Direct values");
    show_debug_message(_GMFILE_);
    show_debug_message(_GMFUNCTION_);
    show_debug_message(_GMLINE_);
    
    show_debug_message("As strings:");
    show_debug_message(string(_GMFILE_));
    show_debug_message(string(_GMFUNCTION_));
    show_debug_message(string(_GMLINE_));
}

panic();
```
The output will be like this:
```
Direct values
gml_GlobalScript_Script1
gml_Script_panic
6
As strings:
@@FILENAME@@
@@FUNC@@
11
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.6.2.162 Runtime v2024.6.1.208

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
Windows

[dfcd491e-ec11-43ca-8fa4-a8a426941d1d](https://api.gamemaker.io/api/github/downloads/dfcd491e-ec11-43ca-8fa4-a8a426941d1d)

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.800.0.593  Runtime v2024.800.0.617
